corehq/form_processor/backends/sql/update_strategy.py

```diff
@@ -287,6 +301,32 @@ def rebuild_from_transactions(self, transactions, rebuild_transaction, unarchive
         if not self.case.modified_on:
             self.case.modified_on = rebuild_transaction.server_date
 
+    def reconcile_transactions_if_necessary(self, xform):
+        if self.case.check_transaction_order():
```

Comment:
It just occurred to me that once this condition is true for a case it will always be true which means we'll end up rebuilding the case on every change. Do you think there's a way of preventing that from happening?

The only thing I could think of was to only query transactions that have taken place since the last rebuild. That would require one extra query to determine when the last rebuild was (or saving a new field on the case) and then adjusting this check accordingly.

Reply:
let me know what you think of the sql change in this bbb1c5d

Reply:
I figured the same query is nicer so it doesn't have to make another http call and since it seems like it's part of the same query somewhat logically. Let me know if you disagree though. Definitely still needs tests